mapvalidator/keys_are.go

```diff
@@ -0,0 +1,61 @@
+package mapvalidator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ tfsdk.AttributeValidator = keysAreValidator{}
+
+// keysAreValidator validates that each map key validates against each of the value validators.
+type keysAreValidator struct {
+	keyValidators []tfsdk.AttributeValidator
+}
+
+// Description describes the validation in plain text formatting.
+func (v keysAreValidator) Description(ctx context.Context) string {
+	var descriptions []string
+	for _, validator := range v.keyValidators {
+		descriptions = append(descriptions, validator.Description(ctx))
+	}
+
+	return fmt.Sprintf("key must satisfy all validations: %s", strings.Join(descriptions, " + "))
+}
+
+// MarkdownDescription describes the validation in Markdown formatting.
+func (v keysAreValidator) MarkdownDescription(ctx context.Context) string {
+	return v.Description(ctx)
+}
+
+// Validate performs the validation.
+func (v keysAreValidator) Validate(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+	elems, ok := validateMap(ctx, req, resp)
+	if !ok {
+		return
+	}
+
+	for k := range elems {
+		request := tfsdk.ValidateAttributeRequest{
+			AttributePath:   req.AttributePath.WithElementKeyString(k),
```

Comment:
Excellent choice here to highlight the whole key/value (at least I think it'll highlight the key too, but regardless, we do not have a way to signal upstream just a map key so this is a good compromise rather than just highlighting the whole map).